src/rust_connection/fd_read_write.rs

```diff
@@ -7,13 +7,27 @@ use std::io::{Error, ErrorKind, IoSlice, Read, Result, Write};
 
 use crate::utils::RawFdContainer;
 
-/// A version of [`std::io::Write`] that also allows sending file descriptors.
-pub trait WriteFD {
-    /// Write a buffer and some FDs into this writer, returning how many bytes were written.
+pub trait Poll {
+    /// Waits or checks for level-triggered read and/or write events
+    /// on the stream.
+    ///
+    /// The first returned boolean specifies whether the stream can
+    /// be read from and the second boolean specifies whether the
+    /// stream can be written into.
+    fn poll(&mut self, read: bool, write: bool, block: bool) -> Result<(bool, bool)>;
```

Comment:
Could/should this use the `BlockingMode` enum from `super`? `foo.poll(true, false, true)` is a bit hard to read.